src/hash_builder/mod.rs

```diff
@@ -90,9 +90,9 @@ impl HashBuilder {
         (self, updates.unwrap_or_default())
     }
 
-    /// Take and return the proofs retained.
-    pub fn take_proofs(&mut self) -> BTreeMap<Nibbles, Bytes> {
-        self.proof_retainer.take().map(ProofRetainer::into_proofs).unwrap_or_default()
+    /// Take and return retained proof nodes.
+    pub fn take_proof_nodes(&mut self) -> ProofNodes {
```

Comment:
this is breaking, but so is the type change, so doesn't matter imo

Reply:
So is https://github.com/alloy-rs/trie/pull/42

Reply:
this is not only a helper struct, but also forces users to revise their usage of proof nodes, because they might rely on them being sorted already like we do in reth